src/java/us/kbase/templates/module_makefile.vm.properties

```diff
@@ -115,11 +115,11 @@ build-test-script:
 	echo 'export KB_AUTH_TOKEN=`cat /kb/module/work/token`' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
 	echo 'echo "Removing temp files..."' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
 	echo 'rm -rf $(WORK_DIR)/*' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'echo "...done removing temp files."' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'echo "Finished removing temp files."' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
 #if($language == "perl")
 	echo 'export PERL5LIB=$$script_dir/../$(LIB_DIR):$$PERL5LIB' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo 'cd $$script_dir/../$(TEST_DIR)' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
-	echo "perl -e 'opendir my ${esc.print("\")}${esc.print("$")}${esc.print("$")}dh, \".\"; my @l = grep { /\\\\.pl${esc.print("\")}${esc.print("$")}${esc.print("$")}/ } readdir ${esc.print("\")}${esc.print("$")}${esc.print("$")}dh; my ${esc.print("\")}${esc.print("$")}${esc.print("$")}ret=0; foreach my ${esc.print("\")}${esc.print("$")}${esc.print("$")}s (@l) { print(\"Running \".${esc.print("\")}${esc.print("$")}${esc.print("$")}s.\"\\\\n\"); system(\"perl\", ${esc.print("\")}${esc.print("$")}${esc.print("$")}s) && do{${esc.print("\")}${esc.print("$")}${esc.print("$")}ret=2; next;}; } exit(${esc.print("\")}${esc.print("$")}${esc.print("$")}ret);'" >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'cd $$script_dir/..' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'prove -lvrm $$script_dir' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
```

Comment:
call perl tests using the perl test harness

Reply:
Sample test output:

```
Run Tests
if [ ! -f /kb/module/work/token ]; then echo -e '\nOutside a docker container please run "kb-sdk test" rather than "make test"\n' && exit 1; fi
bash test/run_tests.sh
Removing temp files...
Finished removing temp files.
/kb/module/test/00_compile.t ...... 
ok 1 - /kb/module/lib/SampleAppClient.pm compiles
ok 2 - /kb/module/lib/SampleApp/SampleAppImpl.pm compiles
ok 3 - /kb/module/lib/SampleApp/SampleAppServer.pm compiles
ok 4 - /kb/module/lib/installed_clients/WorkspaceClient.pm compiles
ok 5 - /kb/module/lib/installed_clients/AssemblyUtilClient.pm compiles
ok 6 - /kb/module/lib/installed_clients/KBaseReportClient.pm compiles
ok 7 - /kb/module/lib/SampleApp.psgi compiles
ok 8 - /kb/module/test/SampleApp_impl.t compiles
ok 9 - /kb/module/test/00_compile.t compiles
1..9
ok
/kb/module/test/SampleApp_impl.t line 31.
ok 1 - require SampleApp::SampleAppImpl;
# Subtest: creating the new SampleAppImpl object
    ok 1 - SampleAppImpl object can be created
    ok 2 - An object of class 'SampleApp::SampleAppImpl' isa 'SampleApp::SampleAppImpl'
    1..2
ok 2 - creating the new SampleAppImpl object
# Subtest: successful run_SampleApp run
Starting run_SampleApp method. Parameters: $VAR1 = {
          'assembly_input_ref' => '48276/1/1',
          'min_length' => 10,
          'workspace_name' => 'test_SampleApp_1601306956268'
        };

Downloading assembly data as FASTA file.
Filtered assembly to 2 contigs out of 3
returning: $VAR1 = {
          'assembly_output' => '48276/1/2',
          'n_contigs_remaining' => 2,
          'n_initial_contigs' => 3,
          'n_contigs_removed' => 1,
          'report_name' => 'report_34f1e646-35af-44ec-9f85-5686ef19d632',
          'report_ref' => '48276/2/1'
        };

    ok 1 - correct number of initial contigs
    ok 2 - correct number of removed contigs
    ok 3 - correct number of remaining contigs
    1..3
ok 3 - successful run_SampleApp run
# Subtest: run_SampleApp errors
Starting run_SampleApp method. Parameters: $VAR1 = {
          'workspace_name' => 'test_SampleApp_1601306956268',
          'assembly_input_ref' => 'fake',
          'min_length' => 10
        };

Downloading assembly data as FASTA file.
    ok 1 - Appropriate error thrown for fake ref object
Starting run_SampleApp method. Parameters: $VAR1 = {
          'workspace_name' => 'test_SampleApp_1601306956268',
          'assembly_input_ref' => 'fake',
          'min_length' => -10
        };

    ok 2 - correct error message for min length param
Starting run_SampleApp method. Parameters: $VAR1 = {
          'assembly_input_ref' => 'fake',
          'workspace_name' => 'test_SampleApp_1601306956268'
        };

    ok 3 - missing min_length param
    1..3
ok 4 - run_SampleApp errors
ok 5 - Test workspace successfully deleted
1..5
ok
All tests successful.
Files=2, Tests=14, 371 wallclock secs ( 0.02 usr  0.01 sys +  1.95 cusr  0.30 csys =  2.28 CPU)
Result: PASS
```

Reply:
This was really a frightening piece of code before!